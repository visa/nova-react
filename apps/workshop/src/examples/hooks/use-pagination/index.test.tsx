/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { fireEvent, render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { UsePaginationExample } from './use-pagination-example';

const examples = [{ Component: UsePaginationExample, title: metaData['use-pagination-example'].title }];

describe('usePagination example', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });

  describe('min/max pagination', () => {
    it('should allow for page change', () => {
      render(<UsePaginationExample />);
      const page5 = screen.getByText('3');
      fireEvent.click(page5);
      expect(page5).toHaveAttribute('aria-current', 'true');
    });
  });
});
