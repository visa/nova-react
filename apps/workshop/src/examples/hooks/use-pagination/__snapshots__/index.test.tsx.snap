// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`usePagination example Behavior and usage should render correctly 1`] = `
<div>
  <nav
    aria-label="minimum and maximum pagination"
    role="navigation"
  >
    <ul
      class="v-pagination v-align-items-center v-flex v-flex-row v-gap-4"
    >
      <li>
        <button
          aria-label="Go to first page"
          class="v-button v-button-small v-button-tertiary v-button-icon"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="v-icon v-icon-visa v-icon-tiny v-icon-arrow-start v-icon-rtl"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="v-icon-primary"
              clip-rule="evenodd"
              d="M2 15.0002V1.00024H4L4 15.0002H2ZM6 8.00025L12.6069 1.00024L14 2.47622L8.78617 8.00025L14 13.5243L12.6069 15.0002L6 8.00025Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to previous page"
          class="v-button v-button-small v-button-tertiary v-button-icon"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="v-icon v-icon-visa v-icon-tiny v-icon-chevron-left v-icon-rtl"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="v-icon-primary"
              clip-rule="evenodd"
              d="M10.6069 1.00049L12 2.47646L6.78617 8.00049L12 13.5245L10.6069 15.0005L4 8.00049L10.6069 1.00049Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="Page 1"
          class="v-button v-button-tertiary"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-current="false"
          aria-label="Page 2"
          class="v-button v-button-tertiary"
        >
          2
        </button>
      </li>
      <li>
        <button
          aria-current="false"
          aria-label="Page 3"
          class="v-button v-button-tertiary"
        >
          3
        </button>
      </li>
      <li
        class="v-pagination-overflow v-align-items-center v-flex"
      >
        <svg
          aria-hidden="true"
          class="v-icon v-icon-visa v-icon-tiny v-icon-option-horizontal"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="v-icon-primary"
            clip-rule="evenodd"
            d="M3 8C3 8.82843 2.32843 9.5 1.5 9.5C0.671573 9.5 0 8.82843 0 8C0 7.17157 0.671573 6.5 1.5 6.5C2.32843 6.5 3 7.17157 3 8ZM9.5 8C9.5 8.82843 8.82843 9.5 8 9.5C7.17157 9.5 6.5 8.82843 6.5 8C6.5 7.17157 7.17157 6.5 8 6.5C8.82843 6.5 9.5 7.17157 9.5 8ZM14.5 9.5C15.3284 9.5 16 8.82843 16 8C16 7.17157 15.3284 6.5 14.5 6.5C13.6716 6.5 13 7.17157 13 8C13 8.82843 13.6716 9.5 14.5 9.5Z"
            fill-rule="evenodd"
          />
        </svg>
      </li>
      <li>
        <button
          aria-current="false"
          aria-label="Page 10"
          class="v-button v-button-tertiary"
        >
          10
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="v-button v-button-small v-button-tertiary v-button-icon"
        >
          <svg
            aria-hidden="true"
            class="v-icon v-icon-visa v-icon-tiny v-icon-chevron-right v-icon-rtl"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="v-icon-primary"
              clip-rule="evenodd"
              d="M5.39309 1.00049L12 8.00049L5.39309 15.0005L4 13.5245L9.21383 8.00049L4 2.47646L5.39309 1.00049Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to last page"
          class="v-button v-button-small v-button-tertiary v-button-icon"
        >
          <svg
            aria-hidden="true"
            class="v-icon v-icon-visa v-icon-tiny v-icon-arrow-end v-icon-rtl"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="v-icon-primary"
              clip-rule="evenodd"
              d="M2 2.47622L7.20919 8.00024L2 13.5243L3.3954 15.0002L10 8.00024L3.3954 1.00024L2 2.47622ZM12 1.00024H14L14 15.0002H12L12 1.00024Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
